packages/gatsby/src/utils/page-data.js

```diff
@@ -35,7 +40,71 @@ const write = async ({ publicDir }, page, result) => {
   await fs.outputFile(filePath, bodyStr)
 }
 
+const getChangedPageDataKeys = (state, cachedPageData) => {
+  if (cachedPageData && state.pageData) {
+    const pageKeys = []
+    state.pageData.forEach((resultHash, key) => {
+      if (!cachedPageData.has(key)) {
+        pageKeys.push(key)
+      } else {
+        const newPageDataHash = resultHash
+        const previousPageDataHash = cachedPageData.get(key)
+
+        if (newPageDataHash !== previousPageDataHash) {
+          pageKeys.push(key)
+        }
+      }
+    })
+    return pageKeys
+  }
+
+  return [...state.pages.keys()]
+}
+
+const collectRemovedPageData = (state, cachedPageData) => {
+  if (cachedPageData && state.pageData) {
+    const deletedPageKeys = []
+    cachedPageData.forEach((_value, key) => {
+      if (!state.pageData.has(key)) {
+        deletedPageKeys.push(key)
+      }
+    })
+    return deletedPageKeys
+  }
+  return []
+}
+
+const checkAndRemoveEmptyDir = (dir, pagePath) => {
+  const hasFiles = fs.readdirSync(path.join(dir, pagePath), `utf8`, true)
```

Comment:
What is the 3rd param of `fs.readdirSync`? I tried checking documentation in https://nodejs.org/api/fs.html#fs_fs_readdirsync_path_options
and it doesn't really have signature with 3 args, so this might be either not used or this could be `withFileTypes` (which I believe we don't make use of anyway as it just change output from array of strings to array of more structured objects with extra details?)

Reply:
This was leftover from debugging, I've removed it